src/maven/background_peaks_update.cpp

```diff
@@ -544,8 +544,13 @@ void BackgroundPeakUpdate::processSlices(vector<mzSlice*>&slices, string setName
 
             //TODO: restructuring, expect that compound is always nullptr
 
-            matchFragmentation(&group, searchableDatabase);
-            //If this is successful, then group.compound will not be nullptr.
+            if (scoringScheme == MzKitchenProcessor::LIPID_SCORING_NAME){
+                MzKitchenProcessor::assignBestLipidToGroup(&group, searchableDatabase, lipidSearchParameters);
+            } else if (scoringScheme == MzKitchenProcessor::METABOLITES_SCORING_NAME) {
+                MzKitchenProcessor::assignBestMetaboliteToGroup(&group, searchableDatabase, mzkitchenMetaboliteSearchParameters);
+            } else {
+                matchFragmentation(&group, searchableDatabase);
+            }
```

Comment:
`matchFragmentation()` is the old method, which will score `PeakGroup` matches the old way, based on whatever the currently selected `scoringScheme` is.